meson.build

```diff
@@ -73,7 +73,15 @@ add_project_arguments ('-DPK_COMPILATION', language: 'c')
 conf = configuration_data()
 conf.set_quoted('DATADIR', join_paths(get_option('prefix'), get_option('datadir')))
 conf.set_quoted('LIBDIR', join_paths(get_option('prefix'), get_option('libdir')))
-conf.set_quoted('SYSCONFDIR', get_option('sysconfdir'))
+if get_option('sysconfdir') == 'etc'
+  pk_sysconfdir = join_paths(get_option('prefix'), get_option('sysconfdir'))
```

Comment:
What is this?!

Reply:
if you compile packagekit form source ... as a normal build without --prefix=/usr ... then it SHOULD install in /usr/local/etc (as prefix is /usr/local and you don't want it going and installing things out-of-prefix that it does not have to).

the problem is it wants to install in "etc" even without the / in front and this then breaks a whole bunch of #define paths (but the meson install actually prepends the / so the files install but then the defines are broken as they contain no leading /). you should be able to install packagekit in a prefix of /usr/local or /opt/whatever for testing/dev/debugging and not mess with the system install of packagkit. in fact the whole sysconfdir handling in packagekit's meson build + defines was rather hairy and it didn't differentiate between looking at existing system files that packagkit is not installing e.g. from pacman and other system packaging systems vs. its own files it installed in sysconfdir. i split these up (sysconfdir vs pk_sysconfdir) so they don't assume it's all the same. as PACKAGE builds will explicitly define sysconfdir as /etc this if clause doesn't trigger as 'etc' without the / in front is meson running as defaults, so this won't affect installs to /usr.

Reply:
Oh dear, well, I guess this fix makes sense in light of that. Is there a Meson bug filed about this?